src/Common/src/System/Resources/ResourceWriter.cs

```diff
@@ -422,7 +412,7 @@ public void Generate()
             bw.Flush();
 
             // Indicate we've called Generate
-            _resourceList = null;
+            _resourceList = null!;
```

Comment:
I agree with the argument you made above for Dispose (we do that in other disposal cases as well), but why is it ok here?

Reply:
Seems should be nullable

Reply:
Same reasoning -- 
1. `Generate()` is analogous to dispose. After it's been called, the type can no longer be used. 
1. This is the only place it is set to null. It is initialized in all constructors. It is not accessible to derived classes. There are no virtual members.
1. For this reason, all entry points already check for null.

Reply:
> For this reason, all entry points already check for null.

If all entry points check for null, then what's the downside to making the field nullable and reflecting reality?  It looks like you'd need to add a Debug.Assert to one internal function, and everything else would just work. No?

Reply:
Sure, I can do this. I need to get myself into the nullable annotation mindset I think 😸